src/BaseGL.js

```diff
@@ -145,9 +148,13 @@ class BaseGL {
    * @param {Array} data.y, y coordinates
    * @param {Array} data.xlabels, labels along the x-axis
    * @param {Array} data.ylabels, labels along the y-axis
+   * @param {Array} legendData, data for the legend
```

Comment:
legendData is an `object`

Reply:
I am also debating if this should be part of the encodings and not input

Reply:
> object

Actually I made legend data an array of objects. Each object contains label, intensity and color.